java/gradle.java/src/org/netbeans/modules/gradle/java/GradleJavaTokenProvider.java

```diff
@@ -99,11 +99,25 @@ private void processSelectedMethod(final Map<String, String> map, Lookup context
         if ((fo != null) && fo.isData()) {
             GradleJavaProject gjp = GradleJavaProject.get(project);
             String className = evaluateClassName(gjp, fo);
-            String selectedMethod = method != null ? className + '.' + method.getMethodName() : className;
+            String selectedMethod = fullyQualifiedMethodName(className, method);
             map.put(SELECTED_METHOD, selectedMethod);
         }
     }
 
+    private String fullyQualifiedMethodName(String className, SingleMethod singleMethod) {
+        if(singleMethod == null) {
+            return className;
+        }
+        StringBuilder sb = new StringBuilder();
```

Comment:
default String Builder capacity will probably be too small to avoid internat buffer realocations. something like

String builder sb = new StringBuilder(className.length() * 3); 

would improve performance most of the time.

Reply:
done